source/common/router/router.cc

```diff
@@ -1539,6 +1539,7 @@ void Filter::UpstreamRequest::onPoolFailure(Http::ConnectionPool::PoolFailureRea
 
 void Filter::UpstreamRequest::onPoolReady(Http::StreamEncoder& request_encoder,
                                           Upstream::HostDescriptionConstSharedPtr host) {
+  ScopeTrackerScopeState scope(&parent_.callbacks_->scope(), parent_.callbacks_->dispatcher());
```

Comment:
IIRC this can happen synchronously during decode headers. I assume double scope tracking works OK? Might be worth a comment?